zenodo/modules/records/query.py

```diff
@@ -30,11 +30,15 @@
 from flask import request
 from invenio_records_rest.query import es_search_factory as _es_search_factory
 
-
 def apply_version_filters(search, urlkwargs):
     """Apply record version filters to search."""
     if request and 'all_versions' in request.values:
-        urlkwargs.add('all_versions', None)
+        if((str(request.values['all_versions']) == "1") or
+                str(request.values['all_versions'].lower()) == "true"):
```

Comment:
1. For readability, `request.values['all_versions']` an be stored in a separate variable (e.g. `all_versions = ...`)
2. `'1'.lower() == '1'`, so the first clause is already covered in the second clause :)
3. The `?all_versions` use-case needs to also be covered (i.e. `all_versions == None`)

Here's how it might look after applying the above:

```suggestion
        all_versions = request.values['all_versions']
        if all_versions is None or str(all_versions).lower() in ("1", "true"):
```